src/BoundaryConditions/continuous_boundary_function.jl

```diff
@@ -77,53 +80,93 @@ function regularize_boundary_condition(bc::BoundaryCondition{C, <:ContinuousBoun
                                                      boundary_func.field_dependencies,
                                                      prognostic_field_names)
 
-    regularized_boundary_func = ContinuousBoundaryFunction{LX, LY, LZ, I}(boundary_func.func,
-                                                                          boundary_func.parameters,
-                                                                          boundary_func.field_dependencies,
-                                                                          indices, interps)
+    regularized_boundary_func = ContinuousBoundaryFunction{LX, LY, LZ, Side}(boundary_func.func,
+                                                                             boundary_func.parameters,
+                                                                             boundary_func.field_dependencies,
+                                                                             indices, interps)
 
     return BoundaryCondition(C, regularized_boundary_func)
 end
 
-@inline boundary_index(i) = ifelse(i == 1, 1, i + 1) # convert near-boundary Center() index into boundary Face() index
+@inline domain_boundary_indices(::LeftBoundary, N) = 1, 1
+@inline domain_boundary_indices(::RightBoundary, N) = N, N + 1
+
+@inline cell_boundary_index(::LeftBoundary, i) = i
+@inline cell_boundary_index(::RightBoundary, i) = i + 1
 
 #####
-##### Kernel functions
+##### Kernel functions for "primitive grid" boundary conditions
 #####
 
-# Return ContinuousBoundaryFunction on east or west boundaries.
-@inline function (bc::ContinuousBoundaryFunction{Nothing, LY, LZ, i})(j, k, grid, clock, model_fields, args...) where {LY, LZ, i}
-    args = user_function_arguments(i, j, k, grid, model_fields, bc.parameters, bc)
+const XBoundaryFunction{LY, LZ, S} = BoundaryCondition{<:Any, <:ContinuousBoundaryFunction{Nothing, LY, LZ, S}} where {LY, LZ, S}
+const YBoundaryFunction{LX, LY, S} = BoundaryCondition{<:Any, <:ContinuousBoundaryFunction{LX, Nothing, LZ, S}} where {LX, LZ, S}
+const ZBoundaryFunction{LX, LY, S} = BoundaryCondition{<:Any, <:ContinuousBoundaryFunction{LX, LY, Nothing, S}} where {LX, LY, S}
 
-    i′ = boundary_index(i)
-
-    return bc.func(ynode(Face(), LY(), LZ(), i′, j, k, grid),
-                   znode(Face(), LY(), LZ(), i′, j, k, grid),
-                   clock.time, args...)
+# Return ContinuousBoundaryFunction on east or west boundaries.
```

Comment:
```suggestion
"Return ContinuousBoundaryFunction on east or west boundaries."
```

Maybe turn these comments into docstring? Equality descriptive but more easily accessible...

Reply:
I can, however these are not user-facing so I don't think we should add docstrings to the documentation library (since we already have a huge number of functions in there).

Reply:
What's the problem of adding docstrings there?